### DISCUSSION TITLE
Install ecole in google colab

### DISCUSSION THREAD

**Discussion by lrushi** [ds4dm/ecole#233] [Questions]
when I tried to install ecole using pip in google colab I encountered the following error

```
Collecting ecole
  Downloading ecole-0.7.2.tar.gz (366 kB)
     |████████████████████████████████| 366 kB 33.3 MB/s 
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
Requirement already satisfied: numpy>=1.4 in /usr/local/lib/python3.7/dist-packages (from ecole) (1.19.5)
Building wheels for collected packages: ecole
  Building wheel for ecole (PEP 517) ... error
  ERROR: Failed building wheel for ecole
Failed to build ecole
ERROR: Could not build wheels for ecole which use PEP 517 and cannot be installed directly
```

I tried many solutions like `!pip install --upgrade pip setuptools wheel` and `!pip install ecole --no-binary :all:` but none of them are working.

Any idea how to install ecole on google colab and use it without errors?

**Comment by AntoinePrv** [comment]
Hi @lrushi,

The `pip` package is currently source only and does not resolve the SCIP dependency. 
This means you need to have both SCIP installed and a C++17 compiler to `pip install ecole`. You can see the compilation error message if you run `pip install -vvv ecole`

You can probably install a compiler and SCIP in Google Colab. However, I would rather try to install `conda`, as the Ecole conda package would be already compiled and downloaded alongside SCIP.

**Reply by lrushi** [reply]
Hi @AntoinePrv 

I installed conda on colab using 2 methods first one from [medium article](https://medium.com/data-professor/how-to-install-conda-on-google-colab-e7bbf9036f76) and second one from this [pip package on github](https://github.com/conda-incubator/condacolab).

Both installed ecole but when I tried to run the example code on the readme of github page of ecole I am encountering some kind of glibc error and with the second conda method I encountered some ImportError from ecole.core.

So If you  can suggest some other method or help fix this issue that would be great

**Reply by AntoinePrv** [reply]
With your first method, could you try to add the following after step 2:
```
!conda update -n base conda
!conda config --add channels conda-forge
!conda config --set channel_priority strict
!conda install -c conda-forge ecole
```
And give us the error message you are getting here.

Additionally, could you give us the output of
```
!conda info
```
